src/projection.jl

```diff
@@ -143,6 +144,11 @@ ProjectTo{P}(::NamedTuple{T, <:Tuple{_PZ, Vararg{<:_PZ}}}) where {P,T} = Project
 # Bool
 ProjectTo(::Bool) = ProjectTo{NoTangent}()  # same projector as ProjectTo(::AbstractZero) above
 
+# Other never-differentiable types
+for T in [:Symbol, :Char, :AbstractString, :RoundingMode, :IndexStyle]
```

Comment:
Not that it is at all a big deal, but prefer tuples for this kind of small literal
Mostly for consistency, but in a few cases it ends up having performance.
```suggestion
for T in (:Symbol, :Char, :AbstractString, :RoundingMode, :IndexStyle)
```